src/ufuncs.c

```diff
@@ -2475,24 +2476,8 @@ static void points_func(char** args, const npy_intp* dimensions, const npy_intp*
       destroy_geom_arr(ctx, geom_arr, i - 1);
       goto finish;
     }
-
-    coord_seq = GEOSCoordSeq_create_r(ctx, 1, n_c1);
-    if (coord_seq == NULL) {
-      errstate = PGERR_GEOS_EXCEPTION;
-      destroy_geom_arr(ctx, geom_arr, i - 1);
-      goto finish;
-    }
-    SINGLE_COREDIM_LOOP_INNER {
-      if (!GEOSCoordSeq_setOrdinate_r(ctx, coord_seq, 0, i_c1, *(double*)cp1)) {
-        errstate = PGERR_GEOS_EXCEPTION;
-        GEOSCoordSeq_destroy_r(ctx, coord_seq);
-        destroy_geom_arr(ctx, geom_arr, i - 1);
-        goto finish;
-      }
-    }
-    geom_arr[i] = GEOSGeom_createPoint_r(ctx, coord_seq);
-    // Note: coordinate sequence is owned by point; if point fails to construct, it will
-    // automatically clean up the coordinate sequence
+    geom_arr[i] = create_point(ctx, *(double*)ip1, *(double*)(ip1 + cs1),
```

Comment:
This is the only part that is hard to follow without a bit more context because of the use of `qip1 + n*cs1` to step over latter dimensions, since that isn't immediately obvious from looking at the implementation of `SINGLE_COREDIM_LOOP_OUTER`.

Suggest adding a comment to explain this a bit more, for future reference.